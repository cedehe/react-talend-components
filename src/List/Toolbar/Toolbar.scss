// Toolbar styles
$tc-list-toolbar-display-color: $silver !default;

.tc-list-toolbar {
	:global(.navbar-text) {
		color: $tc-list-toolbar-display-color;
		font-weight: 400;
		margin-right: 0px;
	}
}
