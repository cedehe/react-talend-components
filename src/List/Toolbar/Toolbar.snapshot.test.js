import React from 'react';
import renderer from 'react-test-renderer';

import Toolbar, { getSubProps } from './Toolbar.component';

jest.mock('react-dom');

const props = {
	onSelectDisplayMode: jest.fn(),
	onSelectSortBy: jest.fn(),
	onFilter: jest.fn(),
	sortBy: [
		{ id: 'id', name: 'Name', selected: true },
		{ id: 'name', name: 'Name' },
	],
	sortDesc: true,
	itemsLength: 17,
	onChangePagination: jest.fn(),
};

describe('Toolbar', () => {
	it('should render', () => {
		// when
		const wrapper = renderer.create(
			<Toolbar {...props} />
		).toJSON();
		
		// then
		expect(wrapper).toMatchSnapshot();
	});
	
	it('should getSubProps', () => {
		// given
		const propTypes = {
			onSelectDisplayMode: 'func',
			sortBy: 'bool',
		};
		
		// when
		const subProps = getSubProps(props, { propTypes });
		
		// then
		expect(subProps.onSelectDisplayMode).toBe(props.onSelectDisplayMode);
		expect(subProps.sortBy).toBe(props.sortBy);
		expect(Object.keys(subProps).length).toBe(2);
	});
	
	it('should render listActions', () => {
		// given
		const tprops = Object.assign({}, props);
		tprops.listActions = [
			{
				label: 'Delete selection',
				icon: 'fa fa-trash-o',
				onClick: jest.fn(),
			},
		];

		// when
		const wrapper = renderer.create(
			<Toolbar {...tprops} />
		).toJSON();
		
		// then
		expect(wrapper).toMatchSnapshot();
	});
	
	it('should render add btn', () => {
		// given
		const tprops = Object.assign({}, props);
		tprops.onClickAdd = jest.fn();
		
		// when
		const wrapper = renderer.create(
			<Toolbar {...tprops} />
		).toJSON();
		
		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render id if provided', () => {
		// given
		const tprops = {
			...props,
			id: 'toolbar',
			onClickAdd: jest.fn(),
			listActions: [
			{
				label: 'Delete selection',
				icon: 'fa fa-trash-o',
				onClick: jest.fn(),
			},
		],
		};
		
		// when
		const wrapper = renderer.create(
			<Toolbar {...tprops} />
		).toJSON();
		
		// then
		expect(wrapper).toMatchSnapshot();
	});
});
