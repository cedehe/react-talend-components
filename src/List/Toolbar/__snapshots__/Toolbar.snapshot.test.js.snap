exports[`Toolbar should render 1`] = `
<div
  className="navbar navbar-default"
  role="toolbar">
  <div
    className="container-fluid">
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Display:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-bars"
                title={null} />
               
              Table
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-list-alt"
                title={null} />
               
              Expanded
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-th-large"
                title={null} />
               
              Tile
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Sort by:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          Name
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
        </ul>
      </li>
      <button
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn btn btn-link"
        disabled={false}
        id={undefined}
        onClick={[Function]}
        onSelect={null}
        role="button"
        type="button">
        DESCENDING
      </button>
    </ul>
    <form
      className="navbar-form navbar-right"
      onSubmit={[Function]}
      role="search">
      <div
        className="form-group">
        <div
          className="input-group">
          <input
            aria-label="Filter"
            className="form-control"
            id={undefined}
            onChange={[Function]}
            placeholder="Filter"
            type="search" />
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          title="Filter"
          type="submit">
          <i
            aria-hidden="true"
            className="fa fa-search" />
          <span
            className="sr-only">
            Filter
          </span>
        </button>
      </div>
    </form>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Show:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          5
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              5
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              10
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              20
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              50
            </a>
          </li>
        </ul>
      </li>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null}>
        <button
          className="btn btn-link"
          disabled={true}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-backward"
            title={null} />
        </button>
        <button
          className="btn btn-link tc-pagination-ctrl-prev btn btn-link tc-pagination-ctrl-prev-default"
          disabled={true}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <span
          className="btn btn-link">
          1
          /
          4
        </span>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-forward"
            title={null} />
        </button>
      </div>
    </ul>
  </div>
</div>
`;

exports[`Toolbar should render add btn 1`] = `
<div
  className="navbar navbar-default"
  role="toolbar">
  <div
    className="container-fluid">
    <ul
      className="nav navbar-nav"
      role={null}>
      <button
        className="navbar-btn btn btn-success"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button">
        <i
          aria-hidden="true"
          className="fa fa-plus"
          title={null} />
        <span>
          Add
        </span>
      </button>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null} />
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Display:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-bars"
                title={null} />
               
              Table
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-list-alt"
                title={null} />
               
              Expanded
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-th-large"
                title={null} />
               
              Tile
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Sort by:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          Name
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
        </ul>
      </li>
      <button
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn btn btn-link"
        disabled={false}
        id={undefined}
        onClick={[Function]}
        onSelect={null}
        role="button"
        type="button">
        DESCENDING
      </button>
    </ul>
    <form
      className="navbar-form navbar-right"
      onSubmit={[Function]}
      role="search">
      <div
        className="form-group">
        <div
          className="input-group">
          <input
            aria-label="Filter"
            className="form-control"
            id={undefined}
            onChange={[Function]}
            placeholder="Filter"
            type="search" />
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          title="Filter"
          type="submit">
          <i
            aria-hidden="true"
            className="fa fa-search" />
          <span
            className="sr-only">
            Filter
          </span>
        </button>
      </div>
    </form>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Show:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          5
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              5
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              10
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              20
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              50
            </a>
          </li>
        </ul>
      </li>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null}>
        <button
          className="btn btn-link"
          disabled={true}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-backward"
            title={null} />
        </button>
        <button
          className="btn btn-link tc-pagination-ctrl-prev btn btn-link tc-pagination-ctrl-prev-default"
          disabled={true}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <span
          className="btn btn-link">
          1
          /
          4
        </span>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-forward"
            title={null} />
        </button>
      </div>
    </ul>
  </div>
</div>
`;

exports[`Toolbar should render id if provided 1`] = `
<div
  className="navbar navbar-default"
  role="toolbar">
  <div
    className="container-fluid">
    <ul
      className="nav navbar-nav"
      role={null}>
      <button
        className="navbar-btn btn btn-success"
        disabled={false}
        id="toolbar-add-btn"
        onClick={[Function]}
        role={null}
        type="button">
        <i
          aria-hidden="true"
          className="fa fa-plus"
          title={null} />
        <span>
          Add
        </span>
      </button>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null}>
        <div
          className="dropdown btn-group btn-group-link">
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-link"
            disabled={false}
            id="toolbar-actions-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button">
            actions
             
            <span
              className="caret" />
          </button>
          <ul
            aria-labelledby="toolbar-actions-dropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                id="toolbar-0-actions-delete-selection"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-trash-o"
                  title={null} />
                Delete selection
              </a>
            </li>
          </ul>
        </div>
      </div>
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor="toolbar-display-mode"
        onSelect={null}>
        Display:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id="toolbar-display-mode"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby="toolbar-display-mode"
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="toolbar-display-mode-table"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-bars"
                title={null} />
               
              Table
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="toolbar-display-mode-large"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-list-alt"
                title={null} />
               
              Expanded
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="toolbar-display-mode-tile"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-th-large"
                title={null} />
               
              Tile
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor="toolbar-sort-by"
        onSelect={null}>
        Sort by:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id="toolbar-sort-by"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          Name
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby="toolbar-sort-by"
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="toolbar-sort-by-item-Name"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id="toolbar-sort-by-item-Name"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
        </ul>
      </li>
      <button
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn btn btn-link"
        disabled={false}
        id="toolbar-sort-order"
        onClick={[Function]}
        onSelect={null}
        role="button"
        type="button">
        DESCENDING
      </button>
    </ul>
    <form
      className="navbar-form navbar-right"
      onSubmit={[Function]}
      role="search">
      <div
        className="form-group">
        <div
          className="input-group">
          <input
            aria-label="Filter"
            className="form-control"
            id="toolbar-filter-input"
            onChange={[Function]}
            placeholder="Filter"
            type="search" />
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          id="toolbar-filter-submit"
          title="Filter"
          type="submit">
          <i
            aria-hidden="true"
            className="fa fa-search" />
          <span
            className="sr-only">
            Filter
          </span>
        </button>
      </div>
    </form>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor="toolbar-pagination-size"
        onSelect={null}>
        Show:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id="toolbar-pagination-size"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          5
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby="toolbar-pagination-size"
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              5
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              10
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              20
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              50
            </a>
          </li>
        </ul>
      </li>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null}>
        <button
          className="btn btn-link"
          disabled={true}
          id="toolbar-nav-to-first"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-backward"
            title={null} />
        </button>
        <button
          className="btn btn-link tc-pagination-ctrl-prev btn btn-link tc-pagination-ctrl-prev-default"
          disabled={true}
          id="toolbar-nav-to-prev"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <span
          className="btn btn-link">
          1
          /
          4
        </span>
        <button
          className="btn btn-link"
          disabled={false}
          id="toolbar-nav-to-next"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <button
          className="btn btn-link"
          disabled={false}
          id="toolbar-nav-to-last"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-forward"
            title={null} />
        </button>
      </div>
    </ul>
  </div>
</div>
`;

exports[`Toolbar should render listActions 1`] = `
<div
  className="navbar navbar-default"
  role="toolbar">
  <div
    className="container-fluid">
    <ul
      className="nav navbar-nav"
      role={null}>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null}>
        <div
          className="dropdown btn-group btn-group-link">
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-link"
            disabled={false}
            id={42}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button">
            actions
             
            <span
              className="caret" />
          </button>
          <ul
            aria-labelledby={42}
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                id={undefined}
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-trash-o"
                  title={null} />
                Delete selection
              </a>
            </li>
          </ul>
        </div>
      </div>
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Display:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-bars"
                title={null} />
               
              Table
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-list-alt"
                title={null} />
               
              Expanded
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              <i
                aria-hidden="true"
                className="fa fa-th-large"
                title={null} />
               
              Tile
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Sort by:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          Name
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              id={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              Name
            </a>
          </li>
        </ul>
      </li>
      <button
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn btn btn-link"
        disabled={false}
        id={undefined}
        onClick={[Function]}
        onSelect={null}
        role="button"
        type="button">
        DESCENDING
      </button>
    </ul>
    <form
      className="navbar-form navbar-right"
      onSubmit={[Function]}
      role="search">
      <div
        className="form-group">
        <div
          className="input-group">
          <input
            aria-label="Filter"
            className="form-control"
            id={undefined}
            onChange={[Function]}
            placeholder="Filter"
            type="search" />
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          title="Filter"
          type="submit">
          <i
            aria-hidden="true"
            className="fa fa-search" />
          <span
            className="sr-only">
            Filter
          </span>
        </button>
      </div>
    </form>
    <ul
      className="nav navbar-nav"
      role={null}>
      <label
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-text"
        htmlFor={undefined}
        onSelect={null}>
        Show:
      </label>
      <li
        className="dropdown"
        style={undefined}>
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle"
          href=""
          id={42}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button">
          5
           
          <span
            className="caret" />
        </a>
        <ul
          aria-labelledby={42}
          className="dropdown-menu"
          role="menu">
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              5
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              10
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              20
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              50
            </a>
          </li>
        </ul>
      </li>
      <div
        active={undefined}
        activeHref={undefined}
        activeKey={undefined}
        className="navbar-btn btn-group"
        onSelect={null}>
        <button
          className="btn btn-link"
          disabled={true}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-backward"
            title={null} />
        </button>
        <button
          className="btn btn-link tc-pagination-ctrl-prev btn btn-link tc-pagination-ctrl-prev-default"
          disabled={true}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <span
          className="btn btn-link">
          1
          /
          4
        </span>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-play"
            title={null} />
        </button>
        <button
          className="btn btn-link"
          disabled={false}
          id={undefined}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-forward"
            title={null} />
        </button>
      </div>
    </ul>
  </div>
</div>
`;
